packages/cdk8s/src/yaml.ts

```diff
@@ -15,6 +20,12 @@ export class Yaml {
    */
   public static save(filePath: string, docs: any[]) {
     // convert each resource to yaml and separate with a '---' line
+    //
+    // NOTE: the `keepUndefined` flag is not supported in the release of yaml
+    //  but is present on the tip of the `master` branch, so this does not work
+    //  currently, but it will when the next release of the yaml package is
+    //  out.
+    // const data = docs.map(r => YAML.stringify(r, {keepUndefined: true})).join('---\n');
```

Comment:
So what's the plan?

Reply:
I think for now, we need to hold on this fix until the `yaml` version has been released to include `keepUndefined`. @eemeli It looks like you're a/the primary maintainer for the `yaml` package. Do you have a timeline for when a version will be released with `keepUndefined`?

Reply:
Maybe we can convert all `undefined`s to `null`s?

Reply:
Soon-ish, esp. as there's clearly a need for it. :) I'm also working on some pretty intensive parser changes, so it'd make sense to wrap up the current state of affairs into a release before those get incorporated. For more specificity, maybe within a week or so?

Reply:
@eemeil, I think within a week should work for my team's timing. Thanks!

@eladb I'll update this PR once the `yaml` version is published. I think that's probably better than working around the issue in the short term since this feature has already been implemented in the library.

Reply:
[`yaml@2.0.0-1`](https://github.com/eemeli/yaml/releases/tag/v2.0.0-1) is now available.

Reply:
Cool, @gabe-l-hart let me know when ready

Reply:
@eemeli Thank you! I'll make the updates and make sure the tests look good